integration-tests/publish.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"github.com/CESARBR/knot-cloud-storage/pkg/data"
+	"github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/logging"
+	"github.com/google/go-querystring/query"
+	"github.com/streadway/amqp"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"time"
+)
+
+func sendRequest(svc *requestInfo) (*http.Response, error) {
+	values, err := query.Values(svc.options)
+	failOnError(err, "Failed to get query values")
+
+	queryString := "?" + values.Encode()
+	client := &http.Client{Timeout: 10 * time.Second}
+	req, err := http.NewRequest(svc.method, svc.url+queryString, bytes.NewBuffer(svc.data))
+	failOnError(err, "Failed to do a HTTP request")
+
+	req.Header.Set("auth_token", svc.authorization)
+	req.Header.Set("Content-Type", svc.contentType)
+
+	return client.Do(req)
+}
+
+
+func publishToken(userTokenValue string, actualTime time.Time) {
+
+	config := setLogger("debug")
+	logrus := logging.NewLogrus(config.Level, false)
+	logger := logrus.Get("Main")
+	logger.Info("Starting storage test")
+
+	conn, err := amqp.Dial("amqp://knot:knot@localhost:5672/")
+	failOnError(err, "Failed to connect to RabbitMQ")
+	defer conn.Close()
+
+	ch, err := conn.Channel()
+	failOnError(err, "Failed to open a channel")
+	defer ch.Close()
+
+	err = ch.ExchangeDeclare(
+		"data.published",
+		"fanout",
+		true,
+		false,
+		false,
+		false,
+		nil,
+	)
+	failOnError(err, "Failed to declare an exchange")
+
+	//payload := setPayload("fbe64efa6c7f717e", []entities.Payload{{SensorID: 1, Value: false}})
+	payload := setPayload("fbe64efa6c7f717e", []entities.Payload{})
+	body, err := json.Marshal(payload)
+	failOnError(err, "Failed to convert to JSON")
+
+	usrToken := userTokenValue
+	startTest(errInvalidToken, validateToken(usrToken, body))
+
+	err = ch.Publish(
+		"data.published",
+		"",
+		false,
+		false,
+		amqp.Publishing{
+			Headers: amqp.Table{"Authorization": usrToken,},
+			ContentType: "application/json",
+			Body: []byte(body),
+		})
+	failOnError(err, "failed to publish data")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI("mongodb://localhost:27017/"))
+	failOnError(err, "Failed to connect to MongoDB")
+	defer func() {
+		if err = client.Disconnect(ctx); err != nil {
+			panic(err)
+		}
+	}()
+
+	database := client.Database("things_db", nil)
```

Comment:
In the same line as my previous comments, separate code that is related to `database` from the test code and the publishing operation.

Reply:
Done. I put the `database`code in a file named `database.go`